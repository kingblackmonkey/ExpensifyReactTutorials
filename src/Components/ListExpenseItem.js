import React from 'react'


import {Link} from 'react-router-dom'
const ListExpenseItem = ({id,description, amount,timePayAt,dispatch})=>(
    <div>
        <li >
        <Link  to= {`/editExpense/${id}`}> <span>{description}</span></Link> <span>{amount}</span> <span>{timePayAt}</span> 
                           
        </li>
   </div>
)


export default ListExpenseItem;