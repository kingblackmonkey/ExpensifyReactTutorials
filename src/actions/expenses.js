import uuidv4 from "uuid/v4"

// add-expense generator
export const addExpenseGenerator = (   
    {   
         description = '',   
          note = '' ,     
           amount = 0,     
            timePayAt = 0    
    } = {} )=>(   
    
        {
        type:"ADD_EXPENSE",
        expense: {
            id: uuidv4(),
            description,
            note,
            amount,
            timePayAt 
        }
       
    }
);

// remove-expense generator 
export const removeExpenseGenerator = ({id} = {})=>(
    {
        type: "REMOVE_EXPENSE",
        id
    }
);
   
// edit-expense generator
export const editExpenseGenerator = ({id, description, amount, note})=>(
    {
        type: "EDIT_EXPENSE",
        expense:{
            id,
            description,
            amount,
            note
        }
    }
);