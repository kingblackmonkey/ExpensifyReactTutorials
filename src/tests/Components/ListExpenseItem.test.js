import React from 'react';
import { shallow} from 'enzyme';
import ListExpenseItem from '../../Components/ListExpenseItem'
import expenses from '../dummyExpenses/dummyExpenses'
test ('should render list expense  item component  ',()=>{
   const wrapper = shallow(
   <ListExpenseItem 
    {...expenses[0]}
   />)   
   expect (wrapper).toMatchSnapshot()
})