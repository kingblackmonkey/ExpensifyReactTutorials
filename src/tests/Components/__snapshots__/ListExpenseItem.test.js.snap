// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render list expense  item component   1`] = `
<div>
  <li>
    <Link
      to="/editExpense/1"
    >
       
      <span>
        redds bill
      </span>
    </Link>
     
    <span>
      12300
    </span>
     
    <span>
      1000
    </span>
  </li>
</div>
`;
